activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb

```diff
@@ -86,6 +86,11 @@ def add_column_options!(sql, options)
             if options[:primary_key] == true
               sql << " PRIMARY KEY"
             end
+
+            if options[:collation].present?
```

Comment:
Do SQL Server and Oracle have support to collation? If not we should not move to abstract adapter.